core/api/src/app/lightning/get-payment-request.ts

```diff
@@ -0,0 +1,77 @@
+import { lookupPaymentByHash } from "./lookup-payment-by-hash"
+
+import { lookupInvoiceByHash } from "./lookup-invoice-by-hash"
+
+import {
+  CouldNotFindLnPaymentFromHashError,
+  CouldNotFindWalletInvoiceError,
+} from "@/domain/errors"
+import { LedgerTransactionType, CouldNotFindTransactionError } from "@/domain/ledger"
+
+import { LedgerService } from "@/services/ledger"
+import { LnPaymentsRepository, WalletInvoicesRepository } from "@/services/mongoose"
+
+export const getPaymentRequestByTransactionId = async ({
+  ledgerTransactionId,
+}: {
+  ledgerTransactionId: LedgerTransactionId
+}): Promise<EncodedPaymentRequest | ApplicationError> => {
+  const ledgerTransaction = await LedgerService().getTransactionById(ledgerTransactionId)
+  if (ledgerTransaction instanceof Error) {
+    return ledgerTransaction
+  }
+
+  const { type, paymentHash } = ledgerTransaction
+  if (!paymentHash) {
+    return new CouldNotFindTransactionError()
+  }
+
+  const isSettledIntraledger =
+    type === LedgerTransactionType.LnIntraLedger ||
+    type === LedgerTransactionType.LnTradeIntraAccount
+  if (isSettledIntraledger || ledgerTransaction.credit > 0) {
+    return getInvoiceRequestByHash({ paymentHash })
+  }
+
+  return getPaymentRequestByHash({ paymentHash })
+}
+
+export const getPaymentRequestByHash = async ({
+  paymentHash,
+}: {
+  paymentHash: PaymentHash
+}): Promise<EncodedPaymentRequest | ApplicationError> => {
+  const lnPayment = await LnPaymentsRepository().findByPaymentHash(paymentHash)
+  if (lnPayment instanceof Error || !lnPayment.paymentRequest) {
```

Comment:
Small nit, could we change this to something like `if ("paymentRequest" in lnPayment && lnPayment.paymentRequest) {` and then early return `lnPayment.paymentRequest` instead of nesting?

Reply:
updated https://github.com/GaloyMoney/galoy/pull/3654/commits/7a33177d706258e78414d39b7297398141e97e67